EIPS/eip-3220.md

```diff
@@ -0,0 +1,121 @@
+|Nickname|Nickname of the chain. Maximum length is 10 characters.|Names such as **Rinkeby** or **Ropsten**.|
+|Chain Type|A meta data for the type of the blockchains such as public or private, permissioned or permissionless| Private:permissioned|
+|Chain Category|A parameter to identify whether this is mainnet, or testnet|"mainnet", "testnet"|
+|Chain IP addresses and Service Provider Id|A set of IP addresses of static nodes that can be contacted to submit a crosschain transaction, and the associated name information for the company providing the service.|125.125.125.125, 10.10.20.20|
+|Name Service Type|The contract naming service available on the blockchain.|The type of service, for instance **ENS**.|
+|Name Service Address|The address of the name service contract on the blockchain.| |
+
+### Crosshchain Identifier Registration Contract
+
+To register crosschian/blockchain id, there should be grace period for comments before it become official and verified.  The registration service should support the following functions
+
+* RegisterBlockchainId
+* ModifyName
+* ModifyParameters
+* VerifyBlockchainId
+* QueryBlockChainId
+* LookupBlockchainId
+* QueryBlockchainOwner
+* QueryBlockchainName(Blockchain-id, NameType)
+* QueryParentBlockchainId
+
+## Rationale
+
+With the success of Bitcoin and Ethereum, various blockchains such as EOS, Ripple, Litecoin, Besu, Wanchain and the like have been developed and are growing at a fast pace.  There are also other private and consortium blockchains such as Hyperledger Fabric, Hyperledger Besu, Stellar, Corda, Quorum that only allow nodes with permitted identities to join the blockchain network.  The growth of public and private blockchains imposes
+challenges for inter-chain interoperability, particularly when these chains are
+heterogeneous and incompatible.
+Enterprise Ethereum Alliance formed Crosschain Interoperability Task Force (CITF) to look into common crosschain problems and solutions. CITF team noticed that there is a lack of unique identifier to charaterise and describe a blockchain. Several proprosals were discussed in EEA Crosschain Interoperability Task Force meetings and discussions.  We have considered various alternative specifications such as using a random unique hex string to represent a blockchain.  The drawback of this method is that the random id can not be used to verify a blockchain's intrinsic identity such as the blockhash of the genesis block.  A second alternative is simply using a genesis blockhash to represent a blockchain id for crosschain operations.  The drawback of this is that this id does not have information about the property of the blockchain and it has problem when a blockchain is forked into two blockchain.  After various discussion, we finally came up with hybrid crosschain identity specification that allows a blockchain to be identified and verified both manually and through computer programs.
```

Comment:
This should all be moved to the motivation section.  The Rationale section is for explaining why you made specific technical choices.  For example, why is the truncated block hash 16 bytes instead of 17 bytes, why is chainId 4 bytes instead of 8 bytes, etc.

Reply:
Modified and moved the descriptive ones to Rational.